src/testdir/test_cd.vim

```diff
@@ -96,10 +96,20 @@ func Test_chdir_func()
   call assert_equal('y', fnamemodify(getcwd(3, 2), ':t'))
   call assert_equal('testdir', fnamemodify(getcwd(1, 1), ':t'))
 
+  " Forcing scope
+  call chdir('.', 'global')
+  call assert_match('^\[global\]', trim(execute('verbose pwd')))
+  call chdir('.', 'tabpage')
+  call assert_match('^\[tabpage\]', trim(execute('verbose pwd')))
+  call chdir('.', 'window')
+  call assert_match('^\[window\]', trim(execute('verbose pwd')))
+
   " Error case
   call assert_fails("call chdir('dir-abcd')", 'E344:')
   silent! let d = chdir("dir_abcd")
   call assert_equal("", d)
+  call assert_fails("call chdir('.', 'illegal')", 'E475:')
+  call assert_fails("call chdir('.', [])", 'E730:')
   " Should not crash
```

Comment:
Can you update the Test_chdir() function in the testdir/test_vim9_builtin.vim file to check for passing null string (test_null_string() or an invalid type (e.g. a List) as argument?